api/v1alpha1/oauth2client_types.go

```diff
@@ -49,6 +49,20 @@ type HydraAdmin struct {
 	// value "off" will force this to be off even if
 	// `--forwarded-proto` is specified
 	ForwardedProto string `json:"forwardedProto,omitempty"`
+
+	// ApiKeySecretRef is an object to define the secret which contains
+	// Ory Network API Key
+	ApiKeySecretRef ApiKeySecretRef `json:"apiKeySecretRef,omitempty"`
+}
+
+// ApiKeySecretRef contains Secret details for the API Key
+type ApiKeySecretRef struct {
+	// Name of the secret containing the API Key
+	Name string `json:"name,omitempty"`
```

Comment:
```suggestion
	Name string `json:"name"`
```
The whole object can be optional, but once given the name of the secret must be supplied.
Moreover, for the object to properly validated by k8s we need to add the annotations like here 

```
	// +kubebuilder:validation:Type=object
	// +nullable
	// +optional
	//
	// Metadata is abritrary data
	Metadata apiextensionsv1.JSON `json:"metadata,omitempty"`
```